typescript/cli/src/config/warp.ts

```diff
@@ -134,15 +140,24 @@ export async function createWarpRouteDeployConfig({
     const isNft =
       type === TokenType.syntheticUri || type === TokenType.collateralUri;
 
-    const mailbox = await detectAndConfirmOrPrompt(
-      async () => {
-        const addresses = await context.registry.getChainAddresses(chain);
-        return addresses?.mailbox;
-      },
-      `For ${chain}, enter the`,
-      'mailbox address',
-      'hyperlane-registry',
-    );
+    // default to the mailbox from the registry and if not found ask to the user to submit one
+    const chainAddresses = await context.registry.getChainAddresses(chain);
+
+    const chainMetadata = await context.registry.getChainMetadata(chain);
+    if (!chainMetadata) {
+      throw new Error(`Metadata not found for chain "${chain}"`);
+    }
```

Comment:
this command did not require there to be chain metadata before your change so not the biggest fan of this only for input address parsing

Reply:
I understand the intent, just dont like that it is a "breaking" change
the CLI will fail for Solana/Cosmos warp routes in other commands so I do not see this as a significant improvement